src/core/client/framework/lib/errors/reporter/sentry.tsx

```diff
@@ -0,0 +1,53 @@
+import * as Sentry from "@sentry/react";
+import React, { FunctionComponent } from "react";
+
+import { ErrorReport, ErrorReporter, User } from "./reporter";
+
+export class SentryErrorReporter extends ErrorReporter {
+  public readonly ErrorBoundry?: FunctionComponent;
+
+  constructor(boundary: boolean, dsn: string) {
+    // Setup the base error reporter.
+    super();
+
+    // Initialize sentry.
+    Sentry.init({ dsn, release: `coral@${process.env.TALK_VERSION}` });
+    Sentry.setTag("domain", window.location.host);
+
+    // Initialize the boundary if enabled.
+
+    // NOTE: (wyattjoh) there should be another way to do this better...
+    if (boundary) {
+      this.ErrorBoundry = function ErrorBoundary({ children }) {
+        return (
+          <Sentry.ErrorBoundary showDialog>{children}</Sentry.ErrorBoundary>
+        );
+      };
+    }
+  }
+
+  public report(err: any): ErrorReport {
+    // Capture and report the error to Sentry.
+    const id = Sentry.captureException(err);
+
+    return { name: "sentry", id };
+  }
+
+  private transformUser(user: User | null): Sentry.User | null {
+    if (!user) {
+      return null;
+    }
+
+    return {
+      id: user.id,
+      username: user.username || undefined,
+      email: user.email || undefined,
```

Comment:
do we need to send email to the error reporter? I'd feel more comfortable if it was just id

Reply:
The email was to aid in responding to users that experience errors, but we can certainly remove it!